src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -0,0 +1,306 @@
+                case "DefaultValue":
+                    RaisePropertyChanged("DefaultValue");
+                    break;
+                case "UsingDefaultValue":
+                    RaisePropertyChanged("UsingDefaultValue");
+                    RefreshPortColors();
+                    break;
+                case "UseLevels":
+                    RaisePropertyChanged("UseLevels");
+                    RaisePropertyChanged(nameof(UseLevelsMenuColor));
+                    break;
+                case "Level":
+                    RaisePropertyChanged("Level");
+                    break;
+                case "KeepListStructure":
+                    RaisePropertyChanged("ShouldKeepListStructure");
+                    RefreshPortColors();
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// UseLevels command
+        /// </summary>
+        public DelegateCommand UseLevelsCommand
+        {
+            get
+            {
+                if (_useLevelsCommand == null)
+                {
+                    _useLevelsCommand = new DelegateCommand(UseLevel, p => true);
+                }
+                return _useLevelsCommand;
+            }
+        }
+
+        private void UseLevel(object parameter)
+        {
+            var useLevel = (bool)parameter;
+            var command = new DynamoModel.UpdateModelValueCommand(
+                Guid.Empty, _node.NodeLogic.GUID, "UseLevels", string.Format("{0}:{1}", _port.Index, useLevel));
+            
+            _node.WorkspaceViewModel.DynamoViewModel.ExecuteCommand(command);
+            RaisePropertyChanged(nameof(UseLevelSpinnerVisible));
+        }
+
+        /// <summary>
+        /// ShouldKeepListStructure command
+        /// </summary>
+        public DelegateCommand KeepListStructureCommand
+        {
+            get
+            {
+                if (_keepListStructureCommand == null)
+                {
+                    _keepListStructureCommand = new DelegateCommand(KeepListStructure, p => true);
+                }
+                return _keepListStructureCommand;
+            }
+        }
+
+        private void KeepListStructure(object parameter)
+        {
+            bool keepListStructure = (bool)parameter;
+            var command = new DynamoModel.UpdateModelValueCommand(
+                Guid.Empty, _node.NodeLogic.GUID, "KeepListStructure", string.Format("{0}:{1}", _port.Index, keepListStructure));
+            
+            _node.WorkspaceViewModel.DynamoViewModel.ExecuteCommand(command);
+        }
+
+        private void ChangeLevel(int level)
+        {
+            var command = new DynamoModel.UpdateModelValueCommand(
+                            Guid.Empty, _node.NodeLogic.GUID, "ChangeLevel", string.Format("{0}:{1}", _port.Index, level));
+
+            _node.WorkspaceViewModel.DynamoViewModel.ExecuteCommand(command);
+        }
+
+        public DelegateCommand MouseLeftButtonDownOnLevelCommand
```

Comment:
does this need to be public for a binding or something? Can we add a ///summary?

Reply:
Yes public for binding... I will add the description